### ISSUE TITLE
Wrong reserved cpus and memory (always 0)

### ISSUE THREAD

**Issue by affo** [docker/swarm#909]
- Cloned swarm.
- Created branch from `master`.
- Added an endpoint to `router.go` with `proxyContainerAndForceRefresh` as handler.
- Built a swarm image from `master` launching `docker build -t myswarm .` in swarm directory.

Now I am launching a swarm cluster with `myswarm` image and everything goes well -- at least
the "custom" endpoint works.  
The problem is that swarm manager doesn't count reserved CPUs and memory (always 0)!

I can run `docker -H tcp://<manager_ip>:4000 run -d -m 1g redis`, but `docker -H tcp://<manager_ip>:4000 info` always outputs `Reserved Memory: 0 B / 16.41 GiB`.

I cannot get if there is a bug in the way `container.Config` is parsed or if I am wrong in my configuration. I discard the possibility to have introduced an error.

Thank you in advance

**Comment by affo** [comment]
I think it is a bug.  
Changed from

```
r += c.Config.Memory
```

to 

```
r += c.Info.HostConfig.Memory
```

in `engine.UsedMemory` and everything works as espected.  
I think the bug should be in method `consolidateResourceFields` in `cluster/config.go`

**Comment by vieux** [comment]
@affo can you write a test to reproduce the issue ?

**Comment by vieux** [comment]
@affo also, what version of docker are you using ?

**Comment by jimmyxian** [comment]
@affo @vieux 
This may related to https://github.com/docker/swarm/issues/908
The latest code in docker have moved the `Memory` filed from `container config` to `host config`

**Comment by affo** [comment]
Sorry for the late answer (different time zones...).
@jimmyxian yes, I think it is related.
@vieux both swarm and docker are built from master, so docker v1.7.0-dev and swarm v0.3.0.

I am trying to write a test.

**Comment by affo** [comment]
I am so sorry for making you lose your time.  Everything seems to be OK now!

However, here's the [(passing) test](https://github.com/4is2cbetter/swarm/blob/fix_reserved/test/integration/config.bats). I can't reproduce the bug anymore because I pulled from upstream on master and rebased. Now I cannot say at which commit I was in the moment I found that bug... If you trust me, I swear that there really was one! :).

Luckily, all ends well :)
Have a nice day and thank you!
